doc/telescope.txt

```diff
@@ -213,6 +213,298 @@ actions.toggle_selection({prompt_bufnr})          *actions.toggle_selection()*
 
 
 
+================================================================================
+                                                          *telescope.previewers*
+
+Provides a Previewer table that has to be implemented by each previewer.
+To achieve this, this module also provides two wrapper that abstract most
+of the work and make it really easy create new previewers.
+  - `previewers.new_termopen_previewer`
+  - `previewers.new_buffer_previewer`
+
+Furthermore, there are a collections of previewers already defined which
+can be used for every picker, as long as the entries of the picker provide
+the necessary fields. The more important once are
+  - `previewers.cat`
+  - `previewers.vimgrep`
+  - `previewers.qflist`
+  - `previewers.vim_buffer_cat`
+  - `previewers.vim_buffer_vimgrep`
+  - `previewers.vim_buffer_qflist`
+
+Previewers can be disabled for any builtin or custom picker by doing
+  :Telescope find_files previewer=false
+
+previewers.Previewer()                                *previewers.Previewer()*
+
+    This is the base table all previewers have to implement. It's possible to
+    write a wrapper for this because most previewers need to have the same
+    keys set.
+    Examples of wrappers are:
+      - `new_buffer_previewer`
+      - `new_termopen_previewer`
+    To create a new table do following:
+      `local new_previewer = Previewer:new(opts)`
+    What `:new` expects is listed below
+
+    The interface provides following set of functions. All of them, besides
+    `new`, will be handled by telescope pickers.
+    - `:new(opts)`
+    - `:preview(entry, status)`
+    - `:teardown()`
+    - `:send_input(input)`
+    - `:scroll_fn(direction)`
+
+    `Previewer:new()` expects a table of as input with following keys:
+      - `setup` function(self): Will be called the first time preview will be
+                              called.
+      - `teardown` function(self): Will be called on cleanup.
+      - `preview_fn` function(self, entry, status): Will be called each time a
+                                                    a new entry was selected.
+      - `send_input` function(self, input): This is meant for
+                                            `termopen_previewer` and it can be
+                                            used to send input to the terminal
+                                            application, like less.
+      - `scroll_fn` function(self, direction): Used to make scrolling work.
+
+
+
+previewers.buffer_previewer_maker({bufnr}, {filepath}, {opts})*previewers.buffer_previewer_maker()*
```

Comment:
Yeah we kinda need a solution for this :|

Reply:
oh! this is actually my first time seeing a long function name inside vimdoc :laughing:
Not sure how to format it.

Reply:
Opened issue upstream i will look into that sometime later.
See https://github.com/tjdevries/tree-sitter-lua/issues/11